tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/FunBuilder.scala

```diff
@@ -41,10 +41,23 @@ trait FunBuilder extends UnsafeFunBuilder {
   /** {{{<<t1, ..., tn>>}}} with a sequence-type. Must be nonempty. */
   def seq(args: TBuilderInstruction*): TBuilderInstruction = buildSeq(args).map { _seq }
 
-  /** [x \in S |-> e] */
-  def funDef(e: TBuilderInstruction, x: TBuilderInstruction, S: TBuilderInstruction): TBuilderInstruction =
-    // funDef args are ordered differently than quantifiers
-    boundVarIntroductionTernary { case (variable, set, expr) => _funDef(expr, variable, set) }(x, S, e)
+  /** [x1 \in S1, ..., xn \in Sn |-> e], must have at least 1 var-set pair */
+  def funDef(e: TBuilderInstruction, varSetPairs: (TBuilderInstruction, TBuilderInstruction)*): TBuilderInstruction =
+    boundVarIntroductionVariadic(_funDef)(e, varSetPairs: _*)
+
+  /**
+   * Alternate call method, where pairs are passed interleaved
```

Comment:
Orthogonal to this PR, but I'd prefer to see the expression repeated on these, otherwise the Scaladoc is pretty useless in auto-completion.

```suggestion
   * [x1 \in S1, ..., xn \in Sn |-> e], must have at least 1 var-set pair
   *
   * Alternate call method, where pairs are passed interleaved.
```

Reply:
IDK how you use Scaladoc in autocompletion, but in my IntelliJ, if you mouseover you can click on the link generated by "@see" and it expands the original comment.

Reply:
But I can go ahead and copy these if you'd like, that's easy.